test/unit/resources/aws_s3_buckets_test.rb

```diff
@@ -0,0 +1,56 @@
+require 'helper'
+
+# MSBB = MockS3BucketsBackend
+# Abbreviation not used outside this file
+
+#=============================================================================#
+#                            Constructor Tests
+#=============================================================================#
+class AwsS3BucketsConstructor < Minitest::Test
+  def setup
+    AwsS3Buckets::BackendFactory.select(AwsMSBB::Basic)
+  end
+
+  def test_constructor_no_args_ok
+    AwsS3Buckets.new
+  end
+
+  def test_constructor_reject_unknown_resource_params
+    assert_raises(ArgumentError) { AwsS3Buckets.new(bla: 'blabla') }
+  end
+end
+
+#=============================================================================#
+#                            Properties
+#=============================================================================#
+class AwsS3bucketsProperties < Minitest::Test
+  def setup
+    AwsS3Buckets::BackendFactory.select(AwsMSBB::Basic)
+  end
+
+  def test_property_bucket_names
+    basic = AwsS3Buckets.new
+    assert_kind_of(Array, basic.bucket_names)
+    assert(basic.bucket_names.include?('bucket_01'))
+    refute(basic.bucket_names.include?(nil))
+  end
+end
+
+#=============================================================================#
+#                               Test Fixtures
+#=============================================================================#
+module AwsMSBB
+  class Basic < AwsBackendBase
```

Comment:
You should also define a test fixture class that is empty, so we can test the negative side of `exist`s.

Reply:
Good call!